### ISSUE TITLE
[Bug]: MvxIosViewPresenter.cs should use .ConfigureAwait(true) _not false_ for virtual methods whose overrides might switch off the main thread

### ISSUE THREAD

**Issue by dsmitchell** [MvvmCross/MvvmCross#4832]
### Description

The following methods can crash because we have no control over subclasses and their overrides:

ShowRootViewController, ShowSplitRootViewController, and ShowPageRootViewController

Note the author of ShowTabBarRootViewController got it right :)

### Steps to Reproduce

Subclass MvxIosViewPresenter and override any of the following methods with actual asynchronous calls:

CloseModalViewControllers, CloseTabBarViewController, CloseSplitViewController

Depending on the implementation, we might actually switch off the main thread (sometimes even inadvertently). And the resulting .ConfigureAwait(false) won't correct the error and we'll get a UIKit crash

### Version with bug

Unknown/Other

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS/MacCatalyst

### Affected platform versions

iOS

### Did you find any workaround?

We cannot workaround because these particular methods are private and not virtual

**Comment by dsmitchell** [comment]
Here's a link to a bad line: https://github.com/MvvmCross/MvvmCross/blob/fc71c2f30b0d6d68d5b37e455f94739ae83e58aa/MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs#L191

**Comment by Cheesebaron** [comment]
@dsmitchell will you make a PR to fix this?

**Comment by entdark** [comment]
@dsmitchell But there is no point in using `ConfigureAwait(true)` unless you want to explicitly mark it.

**Comment by dsmitchell** [comment]
> @dsmitchell But there is no point in using `ConfigureAwait(true)` unless you want to explicitly mark it.

`ConfigureAwait(false)` is a bug in the current code. See the example line I pasted

**Comment by dsmitchell** [comment]
> @dsmitchell will you make a PR to fix this?

Sure, I will try to get around to this

**Comment by dsmitchell** [comment]
> @dsmitchell will you make a PR to fix this?

My first PR is here: https://github.com/MvvmCross/MvvmCross/pull/4834

Please let me know if there's anything else that needs to be done

**Comment by dsmitchell** [comment]
> > > @dsmitchell But there is no point in using `ConfigureAwait(true)` unless you want to explicitly mark it.
> > 
> > 
> > `ConfigureAwait(false)` is a bug in the current code. See the example line I pasted
> 
> My point was having `ConfigureAwait(true)` has no affect and can be removed completely.

That is one option. It is possible the original code was put in place because of compiler warnings to force the developer to understand the implications and pick a behavior. In the case of my PR, rather than remove them I choose to keep the changes minimal...and match what's done in the correctly-behaving method ShowTabBarRootViewController

**Comment by Cheesebaron** [comment]
Closing this as PR has been merged.

I think I added these when fixing CA2007 warnings and just added `ConfigureAwait(false)`, when it probably should have been `ConfigureAwait(true)`.

As @dsmitchell says, adding these explicitly signals the intent, rather than relying on defaults and chance of someone (like me), go in and add `ConfigureAwait(false)` which is not correct in this case. So rather be a bit more explicit with the intent, rather than risking issues in the future.

Thanks for the PR and opening the issue 👍
